internal/server/server_authz_checks.go

```diff
@@ -2,16 +2,66 @@ package server
 
 import (
 	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"crypto/subtle"
+	"encoding/hex"
+	"io"
 	"net/http"
 
 	"github.com/kopia/kopia/internal/auth"
 )
 
+const kopiaSessionCookie = "Kopia-Session-Cookie"
+
+func (s *Server) generateCSRFToken(sessionID string) string {
+	h := hmac.New(sha256.New, s.authCookieSigningKey)
+	io.WriteString(h, sessionID) //nolint:errcheck
```

Comment:
nit; better to panic-on-err if you believe that is an impossible edge / would indicate memory issues. there is nothing productive to be done when that'd happen.

```suggestion
	if _, err := io.WriteString(h, sessionID); err != nil {
	    panic("io.WriteString() failed: " + err.Error())
	}
```

Reply:
done